estimagic/optimization/optimize.py

```diff
@@ -534,7 +534,8 @@ def _single_optimize(
         try:
             algorithm = AVAILABLE_ALGORITHMS[algorithm]
         except KeyError:
-            proposed = propose_algorithms(algorithm, list(AVAILABLE_ALGORITHMS))
+            possibilities = [v.__name__ for v in AVAILABLE_ALGORITHMS.values()]
```

Comment:
Can't we use the keys instead of the `__name__` attribute of the values? This attribute is not always there and might also not be correct. For example, if an algorithm is a partialed function, it does not necessarily have the right `__name__`

Reply:
You are perfectly right (b5fcf5d).